sympy/solvers/solveset.py

```diff
@@ -698,7 +508,7 @@ def _solve_radical(f, symbol, solveset_solver):
     return FiniteSet(*[s for s in result if checksol(f, symbol, s) is True])
 
 
-def _solve_abs(f, symbol):
+def _solve_abs(f, symbol, domain=S.Complexes):
```

Comment:
You should remove the default value and raise an error if the domain is complex. This is solution is valid only on the real domain. On complex domain `Abs(z) = sqrt(Re(z)**2 + Im(z)**2)` and this `Abs` is not invertible, there are uncountably many complex numbers which corresponds to particular absolute value.